tokio/src/net/udp.rs

```diff
@@ -1404,6 +1418,59 @@ impl UdpSocket {
         Poll::Ready(Ok(addr))
     }
 
+    /// Retrieve the sender of the data at the head of the input queue, waiting if empty.
+    ///
+    /// This is equivalent to calling [`peek_from`] with a zero-sized buffer,
+    /// but suppresses the `WSAEMSGSIZE` error on Windows and the "invalid argument" error on macOS.
+    ///
+    /// [`peek_from`]: method@Self::peek_from
+    pub async fn peek_sender(&self) -> io::Result<SocketAddr> {
+        self.io
+            .registration()
+            .async_io(Interest::READABLE, || self.peek_sender_inner())
+            .await
+    }
+
+    /// Retrieve the sender of the data at the head of the input queue,
+    /// scheduling a wakeup if empty.
+    ///
+    /// This is equivalent to calling [`poll_peek_from`] with a zero-sized buffer,
+    /// but suppresses the `WSAEMSGSIZE` error on Windows and the "invalid argument" error on macOS.
+    ///
+    /// # Notes
+    ///
+    /// Note that on multiple calls to a `poll_*` method in the recv direction, only the
+    /// `Waker` from the `Context` passed to the most recent call will be scheduled to
+    /// receive a wakeup.
+    ///
+    /// [`poll_peek_from`]: method@Self::poll_peek_from
+    pub fn poll_peek_sender(&self, cx: &mut Context<'_>) -> Poll<io::Result<SocketAddr>> {
+        self.io
+            .registration()
+            .poll_read_io(cx, || self.peek_sender_inner())
+    }
+
+    /// Try to retrieve the sender of the data at the head of the input queue.
+    ///
+    /// When there is no pending data, `Err(io::ErrorKind::WouldBlock)` is
+    /// returned. This function is usually paired with `readable()`.
+    pub fn try_peek_sender(&self) -> io::Result<SocketAddr> {
```

Comment:
I understand if there's concerns about adding too many methods to `UdpSocket` at once. 

I believe the only one I need to support my use-case is this one, which will be paired with `ready()` and `try_send_to()`.

However, `try_peek_from()` is conspicuously absent from the API so the presence of this method might be confusing, especially if it's the only one.

Reply:
From discussing on the Discord, the resolution here is to just go ahead and add `try_peek_from`. There's no complaints about the number of methods being added, in fact the preference is to try to cover the most possible use-cases.